### DISCUSSION TITLE
Better Error Messages.

### DISCUSSION THREAD

**Discussion by louiss0** [markdoc/markdoc#353] [Ideas]
I have decided to test out markdoc and looked at the error messages and found out that they are not good at all. 
I know this is not easy but attribute validation is one of the strengths of Markdoc. It makes it so it's able to help you find which attribute is improper. But I have noticed that the implementation is not good.

**I get this error when an attribute is not validated correctly.**
 
```
failed to validate:
attribute-value-invalid
```  

**I should instead get this**

```
Error this  attribute {attribute here} is in valid Wrong type 
```

**If I decided to use matches **

```
Error: this attribute should be one of these 1,2,3,4,
```
**In case of words**
```
Error: this attribute should be one of words done, pending, doing,
```
**One time  I got this error from writing a code like this** 

```
{% debug("foo") %}
```

**I got this error**
`failed to validate: child-invalid`
I did not get a line number for this. I could have done the same mistake in multiple lines but I instead I don't get an indicator of where I messed up.    

**Here is the repo where I tried this out**

[Repo From Experiment](https://stackblitz.com/edit/withastro-astro-zdtvav?file=src/pages/index.astro)

**Comment by rpaul-stripe** [comment]
Can you tell me how you are performing the validation? It seems like you are just seeing the error `id`, but Markdoc itself provides very explicit messages. For example, an invalid type shows the [message](https://github.com/markdoc/markdoc/blob/main/src/validator.ts#L209) "Attribute '${key}' must be type of '${t}" and an error on a `matches` rule shows the [message](https://github.com/markdoc/markdoc/blob/main/src/validator.ts#L223) "Attribute '${key}' must match one of ${matches}".

**Reply by louiss0** [reply]
> Can you tell me how you are performing the validation? It seems like you are just seeing the error `id`, but Markdoc itself provides very explicit messages. For example, an invalid type shows the [message](https://github.com/markdoc/markdoc/blob/main/src/validator.ts#L209) "Attribute '${key}' must be type of '${t}" and an error on a `matches` rule shows the [message](https://github.com/markdoc/markdoc/blob/main/src/validator.ts#L223) "Attribute '${key}' must match one of ${matches}".

I now know that this is Astro's fault. Sorry I assumed that it was you guys. I have to reach out to them about this.
